### PR TITLE
docs: add the example of pvc-based ceph cluster on top of bare metal

### PR DESCRIPTION
**Pull request by satoru-takeuchi**
**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [o] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [o] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [o] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [o] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [o] Documentation has been updated, if necessary.
- [o] Unit tests have been added, if necessary.
- [o] Integration tests have been added, if necessary.
- [o] Pending release notes updated with breaking and/or notable changes, if necessary.
- [o] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [o] Code generation (`make codegen`) has been run to update object specifications, if necessary.

[skip ci]

### CODE REVIEW

**Review by subhamkrai — CHANGES_REQUESTED** [review]
commitlint bot is failing.

File: Documentation/ceph-cluster-crd.md [hunk 1]
```diff
@@ -45,12 +45,88 @@ spec:
 ## PVC-based Cluster
 
 In a "PVC-based cluster", the Ceph persistent data is stored on volumes requested from a storage class of your choice.
-This type of cluster is recommended in a cloud environment where volumes can be dynamically created and also
+This type of cluster is recommended to use with CSI drivers which support dynamic volume provisioning.
 in clusters where a local PV provisioner is available.
```
Comment (subhamkrai) [inline Documentation/ceph-cluster-crd.md:49]:
```suggestion
This type of cluster is recommended to use with CSI drivers which support dynamic volume provisioning and also
in clusters where a local PV provisioner is available.
```

**Review by subhamkrai — CHANGES_REQUESTED** [review]

**Review by subhamkrai — APPROVED** [review]
LGTM

**Review by leseb — APPROVED** [review]

File: Documentation/ceph-cluster-crd.md [hunk 2]
```diff
@@ -45,12 +45,88 @@ spec:
 ## PVC-based Cluster
 
 In a "PVC-based cluster", the Ceph persistent data is stored on volumes requested from a storage class of your choice.
-This type of cluster is recommended in a cloud environment where volumes can be dynamically created and also
+This type of cluster is recommended to use with CSI drivers which support dynamic volume provisioning and also in clusters where a local PV provisioner is available.
 in clusters where a local PV provisioner is available.
 
 > **NOTE**: Kubernetes version 1.13.0 or greater is required to provision OSDs on PVCs.
 
 ```yaml
```
Comment (travisn) [inline Documentation/ceph-cluster-crd.md:53]: I'd like to keep the examples at the top of this page simple, and have more complex examples later in the doc. What about this?
- This example should match the cluster-on-pvc.yaml example for aws
- We should add a new section later in the doc for this example that creates local PVs and uses a local provisioner

File: Documentation/ceph-cluster-crd.md [hunk 3]
```diff
@@ -45,12 +45,88 @@ spec:
...
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
```
Comment (travisn) [inline Documentation/ceph-cluster-crd.md:55]: Please add this example yaml in the examples folder, for example named cluster-on-local-pvs.yaml. It's easier to use the examples if we have them in the example folder.

**Review by travisn — CHANGES_REQUESTED** [review]

File: Documentation/ceph-cluster-crd.md [hunk 4]
```diff
@@ -435,6 +435,11 @@ Below are the settings available, both at the cluster and individual node level,
 * `devices`: A list of individual device names belonging to this node to include in the storage cluster.
   * `name`: The name of the device (e.g., `sda`), or full udev path (e.g. `/dev/disk/by-id/ata-ST4000DM004-XXXX` - this will not change after reboots).
   * `config`: Device-specific config settings. See the [config settings](#osd-configuration-settings) below
+
+Host-based cluster only supports raw device and partition.
+
+And below is the settings for PVC-based cluster.
```
Comment (travisn) [inline Documentation/ceph-cluster-crd.md:441]:
```suggestion
Below are the settings for a PVC-based cluster.
```

File: cluster/examples/kubernetes/ceph/cluster-on-local-pvc.yaml [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+                          - rook-ceph-osd
+                      - key: app
+                        operator: In
+                        values:
+                          - rook-ceph-osd-prepare
+                  topologyKey: kubernetes.io/hostname
+          topologySpreadConstraints:
+            - maxSkew: 1
+              # IMPORTANT: If you don't have zone labels, change this to another key such as kubernetes.io/hostname
+              topologyKey: topology.kubernetes.io/zone
```
Comment (travisn) [inline cluster/examples/kubernetes/ceph/cluster-on-local-pvc.yaml:244]: Shall we change this constraint to `kubernetes.io/hostname`? Or maybe we don't need the topology spread constraints for the local storage example if we're only creating OSDs on exactly the number of local PVs that are created.

File: cluster/examples/kubernetes/ceph/cluster-on-local-pvc.yaml [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+            spec:
+              resources:
+                requests:
+                  storage: 10Gi
+              # IMPORTANT: Change the storage class depending on your environment
+              storageClassName: local-storage
+              volumeMode: Block
+              accessModes:
+                - ReadWriteOnce
+        # dedicated block device to store bluestore database (block.db)
```
Comment (travisn) [inline cluster/examples/kubernetes/ceph/cluster-on-local-pvc.yaml:275]: How about if we don't duplicate all the commented settings like metadata/wal? The cluster-on-pvc.yaml example can be followed if needed, and this example can just focus on the local PVs.

File: cluster/examples/kubernetes/ceph/cluster-on-local-pvc.yaml [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+  #      cpu: "200m"
+  #      memory: "200Mi"
+  disruptionManagement:
+    managePodBudgets: true
+    osdMaintenanceTimeout: 30
+    pgHealthCheckTimeout: 0
+    manageMachineDisruptionBudgets: false
+    machineDisruptionBudgetNamespace: openshift-machine-api
+  # security oriented settings
+  # security:
```
Comment (travisn) [inline cluster/examples/kubernetes/ceph/cluster-on-local-pvc.yaml:318]: Similarly, no need for the commented security settings in this example

File: cluster/examples/kubernetes/ceph/cluster-on-local-pvc.yaml [hunk 8]
```diff
@@ -0,0 +1,340 @@
+#################################################################################################################
+# Define the settings for the rook-ceph cluster with common settings for a production cluster on top of bare metal.
+# At least three nodes are required in this example. See the documentation for more details on storage settings available.
+# For example, to create the cluster:
```
Comment (travisn) [inline cluster/examples/kubernetes/ceph/cluster-on-local-pvc.yaml:4]: How about adding a comment at the top that tells the assumptions of the example? We expect three nodes, each with two available disks. The example will certainly need to be modified with their host and device names.

**Review by travisn — CHANGES_REQUESTED** [review]

File: cluster/examples/kubernetes/ceph/cluster-on-pvc.yaml [hunk 9]
```diff
@@ -52,9 +51,6 @@ spec:
       - name: set1
         # The number of OSDs to create from this device set
         count: 3
-        # IMPORTANT: If volumes specified by the storageClassName are not portable across nodes
```
Comment (travisn) [inline cluster/examples/kubernetes/ceph/cluster-on-pvc.yaml:55]: This comment still seems good to keep

Comment (travisn) [reply cluster/examples/kubernetes/ceph/cluster-on-pvc.yaml:55]: The following two lines are also needed

**Comment by satoru-takeuchi** [comment]
@travisn oops, I overlooked the following lines. fixed.

**Comment by satoru-takeuchi** [comment]
@travisn Could you take a look this PR again? I reflected all your suggestion.

**Review by travisn — APPROVED** [review]

**Pull request merged**